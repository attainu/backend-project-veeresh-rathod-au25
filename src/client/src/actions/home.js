import * as ACTIONS from "../utils/actions";
import { store } from "../App";
import Socket from "../utils/socket";

export const getActiveUsers = () => {
  const socket = Socket.getSocket();
  if (socket) {
    store.dispatch({ type: ACTIONS.FETCH_ACTIVE_USERS_PENDING });

    socket.on("activeUsers", (activeUsers) => {
      store.dispatch({ type: ACTIONS.FETCH_ACTIVE_USERS_SUCCESS, payload: activeUsers });
    });
  } else {
    store.dispatch({ type: ACTIONS.FETCH_ACTIVE_USERS_FAILURE, payload: {}, error: "Something went wrong!" });
  }
};
