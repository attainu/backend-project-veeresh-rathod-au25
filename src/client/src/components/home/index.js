import React, { useState } from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import { makeStyles, Modal, Paper, Typography } from '@material-ui/core';

import ActiveUsers from "./active-users";
import PopoverContent from "./popover-content";
import { Body } from "../common";
const MAX = 5;

const useStyles = makeStyles(() => ({
  container: {
    height: "85%",
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  },
  title: {
    display: "flex",
    flex: 0.4,
    alignItems: "center"
  },
  main: {
    display: "flex",
    flex: 0.2,
    alignItems: "center"
  },
  footer: {
    display: "flex",
    flex: 0.4,
    alignItems: "flex-end"
  },
  modalWindow: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  modalCtr: {
    width: "50%",
    overflow: "auto",
    maxHeight: "75%"
  }
}));

const Home = ({ activeUsers, loggedInUser }) => {
  const classes = useStyles();
  const [showDialog, setDialog] = useState(false);

  const isAdmin = get(loggedInUser, "isAdmin", false);
  const handleClose = () => setDialog(false);
  const shownUsers = activeUsers.slice(0, MAX);
  const hiddenUsers = activeUsers.slice(MAX);

  return (
    <Body isLoggedIn={true} admin={isAdmin}>
      <section className={classes.container}>
        <header className={classes.title}>
          <Typography variant="h4">
            Welcome {isAdmin ? "Admin!" : "User!"}
          </Typography>
        </header>
        <main className={classes.main}>
          <ActiveUsers
            users={shownUsers}
            hiddenCount={hiddenUsers.length}
            max={MAX}
            handleMore={() => setDialog(true)} />
          <Modal
            open={showDialog}
            onClose={handleClose}
            className={classes.modalWindow}
          >
            <Paper className={classes.modalCtr}>
              <PopoverContent users={hiddenUsers} />
            </Paper>
          </Modal>
        </main>
        <footer className={classes.footer}>
          <Typography variant="caption">
            Created by: Amruth Skanda, 2020
        </Typography>
        </footer>
      </section>
    </Body>
  );
};

Home.propTypes = {
  activeUsers: PropTypes.array,
  loggedInUser: PropTypes.object
};

export default Home;