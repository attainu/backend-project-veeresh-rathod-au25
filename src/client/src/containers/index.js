export { default as Login } from "./Login";
export { default as Home } from "./Home";
export { default as History } from "./History";
export { default as Registration } from "./Registration";
export { default as Error } from "./Error";
