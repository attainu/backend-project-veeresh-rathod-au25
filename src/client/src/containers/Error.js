import React from "react";

import { Error } from "../components";

const ErrorContainer = (props) => {
  let msg = "Invalid session. Please login again!";
  return <Error message={msg} {...props} />;
};

export default ErrorContainer;