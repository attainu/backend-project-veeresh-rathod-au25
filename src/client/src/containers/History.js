import React from "react";
import PropTypes from "prop-types";

import { History } from "../components";

import * as Hocs from "../hocs";
import { getUserHistory } from "../actions";

export const HistoryContainer = (props) => {
  React.useEffect(() => {
    getUserHistory();
  }, []);

  return <History {...props} />;
};

HistoryContainer.propTypes = {
  users: PropTypes.array
};

const stateMapper = (state) => ({
  users: state.userHistory.payload || []
});

const HistoryWithAuth = Hocs.withAuth(HistoryContainer);
export default Hocs.withStore(stateMapper)(HistoryWithAuth);