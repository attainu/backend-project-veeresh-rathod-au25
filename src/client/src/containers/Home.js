import React from "react";
import PropTypes from "prop-types";

import { Home } from "../components";

import { withStore, withAuth } from "../hocs";
import { getActiveUsers } from "../actions";
import Socket from "../utils/socket";

export const HomeContainer = (props) => {
  React.useEffect(() => {
    if (!Socket.getSocket())
      Socket.create();
    getActiveUsers();
  }, []);

  return <Home {...props} />;
};

HomeContainer.propTypes = {
  activeUsers: PropTypes.object,
  loggedInUser: PropTypes.object
};

const stateMapper = (state) => ({
  activeUsers: Object.values(state.activeUsers.payload),
  loggedInUser: state.login.payload && state.login.payload.user
});

const HomeWithAuth = withAuth(HomeContainer);
export default withStore(stateMapper)(HomeWithAuth);