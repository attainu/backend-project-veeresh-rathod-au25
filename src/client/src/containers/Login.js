import React from "react";
import PropTypes from "prop-types";

import { Login } from "../components";
import { withStore } from "../hocs";
import ReactRouterDom from "react-router-dom";

export const LoginContainer = (props) => {
  const history = ReactRouterDom.useHistory();
  const { loggingIn, loginSuccess } = props;

  React.useEffect(() => {
    if (!loggingIn && loginSuccess) {
      history.push("/app/home");
    }
  }, [loggingIn, loginSuccess]);

  return <Login {...props} />;
};

LoginContainer.propTypes = {
  loggingIn: PropTypes.bool,
  loginSuccess: PropTypes.bool,
  loginFailure: PropTypes.bool,
  signingUp: PropTypes.bool,
  signUpSuccess: PropTypes.bool,
  loggingOut: PropTypes.bool,
  logoutSuccess: PropTypes.bool
};

const stateMapper = (state) => ({
  loggingIn: state.login.loading,
  loginSuccess: state.login.payload,
  loginFailure: state.login.error,
  signingUp: state.signUp.loading,
  signUpSuccess: state.signUp.payload,
  loggingOut: state.logout.loading,
  logoutSuccess: state.logout.payload,
});

export default withStore(stateMapper)(LoginContainer);
