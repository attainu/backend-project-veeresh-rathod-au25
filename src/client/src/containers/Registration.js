import React from "react";
import PropTypes from "prop-types";
import ReactRouterDom from "react-router-dom";

import { Registration } from "../components";
import { withStore } from "../hocs";
import { SIGN_UP_RESET } from "../utils/actions";

export const RegistrationContainer = (props) => {
  const history = ReactRouterDom.useHistory();
  const { signingUp, signUpSuccess, dispatch } = props;

  React.useEffect(() => {
    if (!signingUp && signUpSuccess) {
      history.push("/app");
      setTimeout(() => dispatch({ type: SIGN_UP_RESET }), 3000);
    }
  }, [signingUp, signUpSuccess]);

  return <Registration {...props} />
};

RegistrationContainer.propTypes = {
  signingUp: PropTypes.bool.isRequired,
  signUpSuccess: PropTypes.bool,
  signUpFailure: PropTypes.bool,
  dispatch: PropTypes.func.isRequired
};

const stateMapper = (state) => ({
  signingUp: state.signUp.loading,
  signUpSuccess: state.signUp.payload,
  signUpFailure: state.signUp.error,
});

export default withStore(stateMapper)(RegistrationContainer);