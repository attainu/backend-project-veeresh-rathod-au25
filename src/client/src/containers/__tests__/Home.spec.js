jest.unmock('../../utils/socket');
jest.unmock("../../actions");

import React from "react";
import Home from "../../components/home";
import { HomeContainer } from "../../containers/Home";
import Socket from '../../utils/socket';
import { shallow } from "enzyme";
import * as Actions from "../../actions/home";
Actions.getActiveUsers = jest.fn();
Socket.getSocket = jest.fn().mockReturnValue(undefined);
Socket.create = jest.fn();

describe("CONTAINER: src/containers/Home.js", () => {
  describe("Socket doesn't exist", () => {
    let wrapper;
    beforeEach(() => {
      Socket.getSocket.mockReset();
      Socket.create.mockReset();
      React.useEffect = jest.fn().mockImplementation(f => f());
      wrapper = shallow(<HomeContainer />);
    });

    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should render Home component", () => {
      expect(wrapper.find(Home)).toBeDefined();
    });

    it("should have called getActiveUsers on mount", () => {
      expect(Actions.getActiveUsers).toHaveBeenCalled();
    });

    it("should create socket if socket is not present", () => {
      expect(Socket.create).toHaveBeenCalled();
    });
  });

  describe("when socket existis", () => {
    let wrapper;
    beforeEach(() => {
      Socket.getSocket = jest.fn().mockReturnValue(true);
      Socket.create.mockReset();
      React.useEffect = jest.fn().mockImplementation(f => f());
      wrapper = shallow(<HomeContainer />);
    });
    it("should create socket if socket is not present", () => {
      Socket.getSocket = jest.fn().mockReturnValue(true);
      expect(Socket.create).not.toHaveBeenCalled();
    });
  });
});