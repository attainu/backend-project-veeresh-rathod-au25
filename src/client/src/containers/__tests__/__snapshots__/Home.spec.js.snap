// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONTAINER: src/containers/Home.js Socket doesn't exist should match snapshot 1`] = `ShallowWrapper {}`;
